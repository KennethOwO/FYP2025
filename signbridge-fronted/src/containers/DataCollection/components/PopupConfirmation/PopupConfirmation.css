/* Base styles for the popup confirmation */
.popup-confirmation {
    background: #fff;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
    font-size: 20px;
    display: none;
    z-index: 1001;
    width: 35%;
    border-radius: 5px;
    padding: 20px;
    box-shadow: 2px 5px 10px rgba(0, 0, 0, 0.1);
    max-height: 90vh;
    overflow-y: auto;
}

.popup-confirmation.open {
    display: block;
}

.popup-confirmation-content-header {
    color: #626262;
}

.popup-confirmation-details {
    margin: 0px 20px;
    padding: 20px;
    overflow-x: auto;
    overflow-y: auto;
    box-sizing: border-box;
}

.popup-confirmation-content-button {
    width: 80%;
    margin: 0 auto;
    display: flex;
    justify-content: space-evenly;
    margin-bottom: 10px;
}

@media (max-width: 480px) {
    .popup-confirmation {
        width: 90%;
        font-size: 16px;
        padding: 10px;
    }

    .popup-confirmation-content-header h2 {
        font-size: 18px;
        margin-top: 10px;
    }

    .popup-confirmation-details {
        padding: 20px 0px;
    }


    .popup-confirmation-details .MuiPaper-root {
        max-width: 90vw;
    }

    .popup-confirmation-details .popup-confirmation-td-title {
        width: 35%;
    }

    .popup-confirmation-content-button {
        flex-direction: column;
        /* Stack buttons vertically */
        gap: 10px;
        /* Space between the buttons */
    }

    .popup-confirmation-content-button button {
        width: 100%;
        /* Make buttons take the full width of the container */
    }
}

/* For devices with a width between 481px and 640px (larger mobile screens) */
@media (min-width: 481px) and (max-width: 640px) {
    .popup-confirmation {
        width: 90%;
        font-size: 18px;
        padding: 10px;
    }

    .popup-confirmation-details {
        padding: 20px 10px;
    }

    .popup-confirmation-details .MuiPaper-root {
        max-width: 90vw;
    }

    .popup-confirmation-details .popup-confirmation-td-title {
        width: 30%;
    }

    .popup-confirmation-content-header h2 {
        font-size: 20px;
        margin-top: 10px;
    }

    .popup-confirmation-content-button button {
        padding: 10px 18px;
        font-size: 15px;
    }
}

/* For devices with a width between 641px and 768px (small tablets) */
@media (min-width: 641px) and (max-width: 768px) {
    .popup-confirmation {
        width: 85%;
        font-size: 18px;
    }

    .popup-confirmation-content-header h2 {
        font-size: 20px;
    }

    .popup-confirmation-content-button button {
        padding: 10px 18px;
        font-size: 15px;
    }

    .popup-confirmation-details {
        overflow-y: auto;
        max-width: 90vw;
        padding: 20px 10px;
    }
}

/* For devices with a width between 769px and 1024px (tablets) */
@media (min-width: 769px) and (max-width: 1024px) {
    .popup-confirmation {
        width: 70%;
        font-size: 18px;
        padding: 20px;
    }

    .popup-confirmation-content-header h2 {
        font-size: 22px;
    }

    .popup-confirmation-content-button button {
        padding: 12px 22px;
        font-size: 16px;
    }
}

/* For devices with a width between 1025px and 1280px (small desktops) */
@media (min-width: 1025px) and (max-width: 1280px) {
    .popup-confirmation {
        width: 55%;
    }

    .popup-confirmation-content-header h2 {
        font-size: 22px;
    }

    .popup-confirmation-content-button button {
        padding: 12px 24px;
        font-size: 16px;
    }
}

/* For devices with a width between 1281px and 1440px (larger desktops) */
@media (min-width: 1281px) and (max-width: 1660px) {
    .popup-confirmation {
        width: 45%;
    }

    .popup-confirmation-content-header h2 {
        font-size: 24px;
    }

    .popup-confirmation-content-button button {
        padding: 14px 28px;
        font-size: 18px;
    }
}