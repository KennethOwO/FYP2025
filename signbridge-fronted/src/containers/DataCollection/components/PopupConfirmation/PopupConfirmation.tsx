import React from "react";
import "./PopupConfirmation.css";
import { Button } from "../../../../components/Button/Button";
import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Paper } from "@mui/material";

import ButtonProcessing from "../../../../components/ButtonProcessing/ButtonProcessing";
import { useTranslation } from "react-i18next";

interface PopupConfirmationProps {
    name: string;
    email: string;
    text: string;
    video: string | null;
    isOpen: boolean;
    onClose: () => void;
    onSubmit: any;
    isLoading: boolean;
    setIsLoading: any;
}

const PopupConfirmation: React.FC<PopupConfirmationProps> = ({ name, email, text, video, isOpen, onClose, onSubmit, isLoading, setIsLoading }) => {
    const { t, i18n } = useTranslation();

    return (
        <div className={`popup-confirmation ${isOpen ? "open" : ""}`}>
            <div className="popup-confirmation-content-header">
                <h2>{t("insAreYouSure")}</h2>
            </div>
            <div className="popup-confirmation-content">
                <div className="popup-confirmation-details">
                    <Paper sx={{ overflow: "hidden" }}>
                        <TableContainer
                            sx={{
                                boxShadow: "none",
                                overflowX: "auto", // Allow horizontal scrolling
                            }}
                        >
                            <Table sx={{}}>
                                <TableBody sx={{ backgroundColor: "#f2f2f2" }}>
                                    <TableRow>
                                        <TableCell
                                            className="popup-confirmation-td-title"
                                            sx={{
                                                fontSize: "18px",
                                                wordBreak: "break-word",
                                                border: "none",
                                                width: "20%",
                                                verticalAlign: "top",
                                            }}
                                        >
                                            {t("insName")}
                                        </TableCell>
                                        <TableCell
                                            align="left"
                                            sx={{
                                                fontSize: "16px",
                                                wordBreak: "break-word",
                                                border: "none",
                                                verticalAlign: "top",
                                            }}
                                        >
                                            {name}
                                        </TableCell>
                                    </TableRow>
                                    <TableRow>
                                        <TableCell
                                            className="popup-confirmation-td-title"
                                            sx={{
                                                fontSize: "18px",
                                                wordBreak: "break-word",
                                                border: "none",
                                                width: "20%",
                                                verticalAlign: "top",
                                            }}
                                        >
                                            {t("insEmail")}
                                        </TableCell>
                                        <TableCell
                                            align="left"
                                            sx={{
                                                fontSize: "16px",
                                                wordBreak: "break-word",
                                                border: "none",
                                                verticalAlign: "top",
                                            }}
                                        >
                                            {email}
                                        </TableCell>
                                    </TableRow>
                                    <TableRow>
                                        <TableCell
                                            className="popup-confirmation-td-title"
                                            sx={{
                                                fontSize: "18px",
                                                wordBreak: "break-word",
                                                border: "none",
                                                width: "20%",
                                                verticalAlign: "top",
                                            }}
                                        >
                                            {t("insText")}
                                        </TableCell>
                                        <TableCell
                                            align="left"
                                            sx={{
                                                fontSize: "16px",
                                                wordBreak: "break-word",
                                                border: "none",
                                                verticalAlign: "top",
                                            }}
                                        >
                                            {text}
                                        </TableCell>
                                    </TableRow>
                                    <TableRow>
                                        <TableCell
                                            className="popup-confirmation-td-title"
                                            sx={{
                                                fontSize: "18px",
                                                wordBreak: "break-word",
                                                width: "20%",
                                                border: "none",
                                                verticalAlign: "top",
                                            }}
                                        >
                                            {t("insVideo")}
                                        </TableCell>
                                        <TableCell
                                            align="left"
                                            sx={{
                                                fontSize: "16px",
                                                wordBreak: "break-word",
                                                border: "none",
                                                verticalAlign: "top",
                                            }}
                                        >
                                            {video}
                                        </TableCell>
                                    </TableRow>
                                </TableBody>
                            </Table>
                        </TableContainer>
                    </Paper>
                </div>

                <div className="popup-confirmation-content-button">
                    <Button type="button" onClick={onClose} buttonStyle="btn--reset" buttonSize="btn--large">
                        {t("cancel_btn")}
                    </Button>
                    <ButtonProcessing onClick={onSubmit} buttonStyle="btn-submit" buttonSize="btn-large" isLoading={isLoading} setIsLoading={setIsLoading}>
                        {t("submit_btn")}
                    </ButtonProcessing>
                </div>
            </div>
        </div>
    );
};

export default PopupConfirmation;
