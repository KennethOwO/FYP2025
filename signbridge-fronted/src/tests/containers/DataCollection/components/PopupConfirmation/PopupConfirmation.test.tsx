import React from "react";
import PopupConfirmation from "../../../../../containers/DataCollection/components/PopupConfirmation/PopupConfirmation";
import { cleanup } from "@testing-library/react";
import { create } from "react-test-renderer";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));

describe("Test PopupConfirmation", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  const props = {
    name: "John Dope",
    email: "JohnDoe@gmail.com",
    text: "This is john doe testing",
    video: "demo-1222222.mp4",
    isOpen: true,
    onClose: jest.fn(),
    onSubmit: jest.fn(),
    isLoading: false,
    setIsLoading: jest.fn(),
  };
  it("should render correctly", () => {
    const tree = create(<PopupConfirmation {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
