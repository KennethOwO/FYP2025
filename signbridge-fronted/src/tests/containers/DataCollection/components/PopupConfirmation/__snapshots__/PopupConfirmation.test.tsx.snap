// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PopupConfirmation should render correctly 1`] = `
<div
  className="popup-confirmation open"
>
  <div
    className="popup-confirmation-content-header"
  >
    <h2>
      insAreYouSure
    </h2>
  </div>
  <div
    className="popup-confirmation-content"
  >
    <div
      className="popup-confirmation-details"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-175ib4q-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          className="MuiTable-root css-rqglhn-MuiTable-root"
          role={null}
        >
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root css-ys9ggv-MuiTableRow-root"
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
              >
                <b>
                  insName
                </b>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-3urw24-MuiTableCell-root"
              >
                John Dope
              </td>
            </tr>
            <tr
              className="MuiTableRow-root css-2qi3sl-MuiTableRow-root"
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
              >
                <b>
                  insEmail
                </b>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
              >
                JohnDoe@gmail.com
              </td>
            </tr>
            <tr
              className="MuiTableRow-root css-ys9ggv-MuiTableRow-root"
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
              >
                <b>
                  insText
                </b>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
              >
                This is john doe testing
              </td>
            </tr>
            <tr
              className="MuiTableRow-root css-1yq2m0q-MuiTableRow-root"
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
              >
                <b>
                  insVideo
                </b>
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
              >
                demo-1222222.mp4
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      className="popup-confirmation-content-button"
    >
      <button
        className="btn btn--reset btn--large"
        data-testid="button-btn--reset"
        onClick={[MockFunction]}
        type="button"
      >
        cancel_btn
      </button>
      <button
        className="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default btn-submit btn-large"
        data-testid="submit_btn2"
        disabled={false}
        onClick={[Function]}
        style={
          {
            "fontSize": "20px",
            "height": "auto",
            "position": "relative",
          }
        }
        type="button"
      >
        <span>
          submit_btn
        </span>
      </button>
    </div>
  </div>
</div>
`;
